src/appleseed/renderer/kernel/rendering/renderercomponents.cpp

```diff
@@ -347,6 +355,47 @@ bool RendererComponents::create_pixel_renderer_factory()
 
         return true;
     }
+    else if (name == "texture")
+    {
+        if (m_sample_renderer_factory.get() == nullptr)
+        {
+            RENDERER_LOG_ERROR("cannot use the texture controlled pixel renderer without a sample renderer.");
+            return false;
+        }
+
+        ParamArray tex_sampler_params = get_child_and_inherit_globals(m_params, "texture_controlled_pixel_renderer");
+        const string tex_path = tex_sampler_params.get_optional<string>("file_path", "");
+
+        if (tex_path.empty())
+        {
+            RENDERER_LOG_ERROR("no texture path was specified for the texture controlled pixel renderer.");
+            return false;
+        }
+
+        // Create a new dedicated ImageCache instead of using the global ImageCache.
+        // This is to prevent errors during ImageBuf access if the file behind tex_path was overwritten on disk.
+        ImageCache* image_cache = ImageCache::create();
```

Comment:
You should use facilities in `foundation/image/` for image I/O instead of using OIIO directly.

Reply:
Hey, so the reason why I want to use OIIO directly here, is because I want to resize the supplied texture to exactly match the frame dimensions and I found that the easiest way to do that would be with OIIO ImageBuf and ImageBufAlgo::resize(). I could use the GenericImageFileReader in `foundation/image/` but this would return type Image and I guess the only way to get an ImageBuf from that is to then loop over each pixel of the Image to get each pixel value and set it in the ImageBuf and then proceed with the resize. That seems like a bit of a detour that I would rather avoid.
Maybe we already have some custom resizing functionality implemented somewhere that I am not seeing?

Reply:
Ah, I see.

> Maybe we already have some custom resizing functionality implemented somewhere that I am not seeing?

I don't think we do.

Another remark: Image loading should be done by a method [1] in `TextureControlledPixelRendererFactory`; `RendererComponents` is only about creating and connecting components together, it shouldn't contain other logic.

[1] Since you want to be able to react to I/O errors, it's best to do this in a separate method vs doing it in the constructor...